dulwich/config.py

```diff
@@ -524,6 +524,13 @@ def default_backends(cls):
 
         if "GIT_CONFIG_NOSYSTEM" not in os.environ:
             paths.append("/etc/gitconfig")
+            if sys.platform == "win32":
+                paths.extend(
+                    [
+                        "C:/Program Files/Git/etc/gitconfig",
```

Comment:
I'm not a big fan of peeking at another applications' directory.

https://git-scm.com/book/en/v2/Getting-Started-First-Time-Git-Setup suggests C:\Documents and Settings\All Users\Application Data\Git\config and C:/ProgramData/Git/config instead. Can we use those?

Reply:
The `Application Data\...` path is listed as the default in XP so I wasn't sure if it was worth including.

My concern here is that the listed paths in the git docs aren't up to date with the current Git for Windows installer. If you download the latest version from https://git-scm.com/download/win and do a fresh installation with all of the default settings, it puts the system config (containing `core.autocrlf=true`) in `C:/Program Files/Git/etc/gitconfig` and does not generate any `C:/ProgramData/Git/` directory at all.

I understand that dulwich is intended to work without relying on CGit, but in the event that I do have a default CGit installation on windows, I would expect dulwich to read the system `autocrlf` setting from the default config location.

Reply:
The default system config location was changed away from `C:/ProgramData` here: https://github.com/git-for-windows/git/pull/2358

Reply:
I'm not necessarily opposed to this merge proposal, but I'd like to have a concise strategy - if we change the behaviour now people will start depending on it.

The other question is what to do in the case that we don't have a CGit installation. Should we have a Dulwich-specific global configuration file? What does e.g. libgit2 read?

Reply:
libgit2 checks to see if `git.exe` is in the current PATH, or if certain registry keys set by msysgit exist, and then sets the system config dir based on that path or installation dir

https://github.com/libgit2/libgit2/blob/508361401fbb5d87118045eaeae3356a729131aa/src/win32/findfile.c#L168

and they also check for the `%PROGRAMDATA%` (`C:/ProgramData/Git`) dir mentioned in the git docs

Reply:
They document the behavior for `git_config_find_system()` as
```
 * If /etc/gitconfig doesn't exist, it will look for
 * %PROGRAMFILES%\Git\etc\gitconfig.
 ```

Reply:
Not sure that dulwich needs to go as far as checking registry keys, but looking for CGit in the PATH seems like a reasonable solution to me

Reply:
I'm a little annoyed by the move away from %PROGRAMDATA%, since that was an easy to find location for a bit of config that is shared between multiple Git applications/libraries. Anyway, that's hard to revert now. This way we'll end up searching for msysgit2, C Git for Windows, libgit2 installations and the behaviour becomes more opaque and unpredictable to the user.

I do think we should rely on registry settings if we're going to find the C Git installation. E.g. on localized systems the paths might vary. It might also be nice to e.g. provide a global list of paths that will be tried (a list of callables) so that Dulwich users can register/tweak git installation find functions - but not necessary for this PR.

Reply:
after 0530d81 behavior in this PR should be consistent with libgit2 (try all of %PROGRAMDATA%, CGit from PATH, CGit from registry)